tests/functionaltests/functionaltest-api.h

```diff
@@ -385,4 +402,88 @@
 #endif
 #define EXPECT_OBJCNE_MSG(val1, val2, format, ...)                                   \
     FunctionalTestLog::SetUserComment(GTestLogPrivate::Format(format, __VA_ARGS__)); \
-    EXPECT_OBJCNE(val1, val2);
\ No newline at end of file
+    EXPECT_OBJCNE(val1, val2);
+
+// clang-format off
+#pragma push_macro("TEST")
+#define TEST(className, methodName)  \
+    \
+    class className##_##methodName { \
+        TEST_CLASS(className##_##methodName) \
+    \
+        TEST_METHOD(Test); \
+    };  \
+    \
+    void className##_##methodName::Test()
+
+#pragma push_macro("TEST_F")
+#define TEST_F(className, methodName)                  \
+/* This assumes that there already exists a className type with SetUp and TearDown methods to call */ \
+    \
+    class className##_##methodName : className { \
+        TEST_CLASS(className##_##methodName) \
+    \
+        TEST_METHOD(Test); \
+        TEST_METHOD_SETUP(_Setup); \
+        TEST_METHOD_CLEANUP(_Teardown); \
+        \
+        /* TODO: fix up inheritance to not use GTestClasses if not needed. */ \
+        virtual void TestBody() { \
+            Test(); \
+        } \
+    }; \
+    \
+    bool className##_##methodName::_Setup() { \
+        SetUp(); \
+        return true; \
+    } \
+    bool className##_##methodName::_Teardown() { \
+        TearDown(); \
+        return true; \
+    } \
```

Comment:
needed for bool conversion. Maybe should guard differently like with a try/catch to indicate success?